src/layers/mlp_llama.cpp

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2023 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ============================================================================
+#include "mlp_llama.h"
+#include "layers_mlp.h"
+
+#include <unordered_map>
+
+namespace xft {
+
+void invokeMLPLLaMA(DataType dt, int numTokens, int hiddenSize, int intermediateSize, void *output, int outputStride,
+        const void *input, int inputStride, const void *gateWeight, const void *upWeight, const void *downWeight) {
+    static std::mutex mutex;
+    std::lock_guard<std::mutex> lock(mutex);
+
+    if (dt == DataType::bf16) {
+        static std::unordered_map<std::string, std::tuple<DecoderContext *, LlamaMLP<bfloat16_t> *>> llama_mlp_hub;
+
+        // create hash key
+        std::stringstream weights_addr;
+        weights_addr << gateWeight << "_" << upWeight << "_" << downWeight;
+        std::string llama_mlp_key
```

Comment:
I think the address is enough, do we really need to add the size info in the key?

Reply:
Done